### DISCUSSION TITLE
Check which column is throwing unable to cast exception

### DISCUSSION THREAD

**Discussion by landetus** [JoshClose/CsvHelper#1904] [Q&A]
Hi everyone, I'm reading a file and I'm getting an "Unable to cast object of type 'System.Int32' to type 'System.Nullable`1[System.Decimal]'" exception. 

It's not caught by the TypeConverterException, where I can check which column has thrown the exception through "ex.MemberMapData.Names[0]", instead it's caught by the CsvHelperException where I've been unable to check which column it's throwing the error. Is there a way to check which specific column is throwing it?

I'm using CsvHelper 27.1.1, and the solutions found in stackoverflow seem to be outdated like https://stackoverflow.com/questions/21609348/in-csvhelper-how-to-catch-a-conversion-error-and-know-what-field-and-what-row-it

Thanks in advance and sorry for my english.
Ricardo.

**Comment by JoshClose** [comment]
The exception message should have a lot of information in it. Can you post that here?

**Reply by landetus** [reply]
Of course, here you have the ex.Message:

> An unexpected error occurred.
> IReader state:
>    ColumnCount: 0
>    CurrentIndex: 30
>    HeaderRecord:
> ["Id Tipo Problema","Tipo Problema","Subtipo","Clasificación","Época","Año","Región","Provincia","Identificador ST 1","Nombre ST 1","Identificador CT 1","Nombre CT 1","Número Línea 1","Nombre Línea 1","Número Problema en Línea 1","Número Trafo 1","Tensión","Municipio","Ámbito","Identificador ST 2","Nombre ST 2","Número Línea 2","Nombre Línea 2","Número Trafo 2","Descripción del Fallo","Comentario","Longitud Aéreo","Longitud Subterráneo","Nº CTs Intemperie","Nº CTs Interior","Nº Disparos Anual","Tiempo Reposición","Potencia instalada interrumpida @ 100% demanda máx en zona urbana","Potencia instalada interrumpida @ 100% demanda máx en zona semiurbana","Potencia instalada interrumpida @ 100% demanda máx en zona rural concentrada","Potencia instalada interrumpida @ 100% demanda máx en zona rural dispersa","Potencia instalada interrumpida @ 80% demanda máx","Potencia interrumpida @ 100% demanda máx","Potencia interrumpida @ 80% demanda máx","Potencia interrumpida @ 100% demanda máx en zona urbana","Potencia interrumpida @ 100% demanda máx en zona semiurbana","Potencia interrumpida @ 100% demanda máx en zona rural concentrada","Potencia interrumpida @ 100% demanda máx en zona rural dispersa","ENS en zona urbana","ENS en zona semiurbana","ENS en zona rural concentrada","ENS en zona rural dispersa","Riesgo  Electrificación","Riesgo  Generación","Penalización por Incentivo de pérdidas","Pérdidas","Grado de Sobrecarga","Máxima Demanda (saturación)","Horas > 100%","Tensión Extrema","Máxima Demanda (tensión)","Máxima Demanda Admisible (tensión)","Horas fuera de límites","Año  Máximo"]
> IParser state:
>    ByteCount: 0
>    CharCount: 18802
>    Row: 68
>    RawRow: 68
>    Count: 59
>    RawRecord:
> 33;Disparos Frecuentes;;;Anual;2021;Norte;AVILA;67;ST CH CHORRO;;;1;EL CHORRO-BARCO;;;;;;;;;;;;;;;;; 5 uds ;;;;;;; 11,00 MW ;;;;;;;;;;;;;;;;;;;;;

And here it is the ex.InnerException.Message:

> Unable to cast object of type 'System.Int32' to type 'System.Nullable`1[System.Decimal]'.

Do you want any other part of the exception? Or can you tell me how to print all that you need?

Thanks in advance Josh.
Ricardo.
